### ISSUE TITLE
Workload feature band fallback doesn't work with rollback files

### ISSUE THREAD

**Issue by dsplaisted** [dotnet/sdk#23402]
When updating workload versions using a [rollback file](https://github.com/dotnet/sdk/blob/main/documentation/general/workload-rollback.md), it will not work correctly if any of the workload manifests are using a different feature band (via the fallback feature).

This is because the rollback file specifies the manifest ID and version, but to find the package that includes the manifest, the feature band is added to the manifest ID to create the manifest package ID.  When applying the rollback file, the SDK's current feature band will be used when trying to download the manifest package, even if the original manifest used a different feature band.

To fix this, we should update the format of the rollback file to include the feature band that the manifest is branded for, and use that when constructing the package ID to download.

- [ ] Update --print-rollback to include the feature band
- [ ] Update the rollback reader to include parse the manifest version and use during update of the manifest
- [ ] When applying rollback, uninstall existing manifest before installing new one as necessary (otherwise a newer band manifest could be left on disk and the rollback wouldn't actually take effect)
- [ ] If no band found in the rollback, default to either the current SDK band or the current manifest band (not sure which is preferred)
- [ ] Inform our partners who may have hardcoded rollback files that they could be impacted

https://github.com/dotnet/sdk/blob/22c4860dcb2cf6b123dd641cc4a87a50380759d5/src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs#L206

**Comment by Redth** [comment]
So would this allow us to do something like:

```
{
  "microsoft.net.sdk.android" : { 
    "version" : "31.0.200-preview.14.106",
    "sdkBand" : "6.0.200"
  },
  "microsoft.net.sdk.ios" : {
    "version" : "15.2.302-preview.14.91",
    "sdkBand" : "6.0.200"
  },
  "microsoft.net.sdk.maccatalyst" : {
    "version" : "15.2.302-preview.14.91",
    "sdkBand" : "6.0.200"
  },
  "microsoft.net.sdk.macos" : {
    "12.1.302-preview.14.91",
    "sdkBand" : "6.0.200"
  },
  "microsoft.net.sdk.maui" : {
    "6.0.200-preview.14.5055",
    "sdkBand" : "6.0.200"
  },
  "microsoft.net.sdk.tvos" : {
    "version" : "15.2.302-preview.14.91",
    "sdkBand" : "6.0.200"
  }
  "microsoft.net.workload.mono.toolchain" : {
     "version" : "6.0.3"
  },
  "microsoft.net.workload.emscripten" : "6.0.2" : {
     "version" : "6.0.1",
     "sdkBand" : "6.0.100"
  }
}
```

Where for example we are running the command against the 6.0.300 SDK, but specifying a `6.0.200` band for a few packages (so that it will find the correct package id with `*Manifest-6.0.200` and perhaps even another version like `6.0.100` for another package?   Also in this example, `6.0.300` is implied for the ToolChain workload manifest.

Is this an accurate understanding of the intent?

**Comment by gkulin** [comment]
@Redth this should be fixed in #24545  it was just merged into release/6.0.3xx
